### ISSUE TITLE
uvicorn 0.2 not compatible with django channels

### ISSUE THREAD

**Issue by ChillarAnand** [Kludex/uvicorn#106]
daphe server for a project(avilpage) can be started with

```
daphne avilpage.asgi:application                                         
```

uvicorn fails with 

```
->uvicorn avilpage.asgi:application
Usage: uvicorn [OPTIONS] APP

Error: Error loading ASGI app. Could not import module "avilpage.asgi".
```

eventhough 

```
python -c 'import avilpage.asgi'                     
```
works fine without any errors.

**Comment by lovelydinosaur** [comment]
I’d suggest having a look at the loading code in main.py to see what’s going wrong. If you can provide a *minimal* reproducible example that’d be good. My be an issue with PYTHONPATH being set incorrectly or something along those lines.

**Comment by lovelydinosaur** [comment]
I’m AFK right now, but this shouldn’t be too tough to resolve with a bit of debugging. Loading the app is a small bit of code. Pull requests very welcome.

**Comment by lovelydinosaur** [comment]
Oh I guess we need to split on the “.” for the import, probably not handling that correctly yet.

Temporary workaround: ensure the application is at the top-level of the module. Or run using gunicorn (see the docs)

**Comment by lovelydinosaur** [comment]
Setting PYTHONPATH would resolve your issue.
Resolved in 0.2.4, where we automatically add the current directory to the PYTHONPATH.
(I've checked this against how `daphne` handles the app import)

**Comment by ChillarAnand** [comment]
Working fine with `uvicorn==0.2.4`. Thanks.
